mini-projects/00-jq-rs/src/filters.rs

```diff
@@ -0,0 +1,499 @@
+use core::panic;
+use serde_json::Value;
+
+use clap::{Arg, Command};
+use std::env;
+use std::fs::File;
+use std::io::BufReader;
+
+#[derive(Debug, PartialEq)]
+pub enum Filter {
+    Identity,
+    ObjectIdentifier(String),
+    ArrayIndex(isize),
+    ArraySlice(Option<isize>, Option<isize>),
+    ArrayIterator,
+    Add,
+    Length,
+    Del(Box<Filter>),
```

Comment:
This is an excellent way to define the filters, especially for the Del to leverage information you already have

Reply:
huge fan of this clean interface btw